File: app/src/main/java/com/example/pam_project/database/lists/ListDao.java
package com.example.pam_project.database.lists;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Transaction;
import androidx.room.Update;

import com.example.pam_project.database.relationships.ListsWithTasks;

import io.reactivex.Flowable;

@Dao
public interface ListDao {
    @Query("SELECT * FROM Lists WHERE id =:id")
    Flowable<ListEntity> getListById(final long id);

    @Insert
    void insertList(final ListEntity list);

    @Update
    void updateList(final ListEntity list);

    @Delete
    void deleteList(final ListEntity list);

    @Transaction
    @Query("SELECT * FROM Lists WHERE id =:id")
    Flowable<ListsWithTasks> getListsWithTasks(final long id);
}

File: app/src/main/java/com/example/pam_project/database/lists/ListDao.kt
package com.example.pam_project.database.lists

import androidx.room.*
import com.example.pam_project.database.relationships.ListsWithTasks
import io.reactivex.Flowable

@Dao
interface ListDao {
    @Query("SELECT * FROM Lists WHERE id =:id")
    fun getListById(id: Long): Flowable<ListEntity?>

    @Insert
    fun insertList(list: ListEntity?)

    @Update
    fun updateList(list: ListEntity?)

    @Delete
    fun deleteList(list: ListEntity?)

    @Transaction
    @Query("SELECT * FROM Lists WHERE id =:id")
    fun getListsWithTasks(id: Long): Flowable<ListsWithTasks?>
}